### ISSUE TITLE
could I get a name from the yas::shared_buffer?

### ISSUE THREAD

**Issue by DarwinRepo** [niXman/yas#127]
Just like this :

yas::shared_buffer = save "MyNameIsBuffer" ... other data;

-- shared_buffer has a member funciton to get the Name = "MyNameIsBuffer"？

**Comment by DarwinRepo** [comment]
Can I get its name from a serialized yas::shared_buffer？

**Comment by niXman** [comment]
name of serialized YAS_OBJECT ?

**Comment by DarwinRepo** [comment]
yes

**Comment by niXman** [comment]
no, you can't.

but you can provide some ID on save stage, then read it.

smth like this:
```cpp
struct mytype {
   std::string_view ID; // must be first
   int m0, m1;
};
```
then you can read it:
```cpp
yas::intrusive_buffer buf = ...
yas::mem_istream is{buf};
yas::binary_iarchive<...> ia{is};

std::string ID;
ia & ID;

if ( ID == "some string ID" ) {
}
```

**Comment by niXman** [comment]
I didn't check the code, use it as pseudocode

**Comment by DarwinRepo** [comment]
ok , another question : could I use std::move to operate a yas::shared_buffer?

**Comment by DarwinRepo** [comment]
if i use like this:
yas::shared_buffer data; // have a lot of data
std::string name = "name";
auto target_data = yas::save<flags>( YAS_OBJECT(“data” ，name ,  data ) );
will the function "save" is copy a lot of data?

**Comment by niXman** [comment]
> could I use std::move to operate a yas::shared_buffer?

yes, between two shared_buffer.

> will the function "save" is copy a lot of data?

will be copied all the data from `name` and `data`.

do you want to add some ID to the already serialized `data`? if yes - please explain why the `ID` and the `data` must be in one serialized buffer? how are you planning to use `target_data` later ?

**Comment by DarwinRepo** [comment]
send it to another process？ and  get the ID first step.

**Comment by niXman** [comment]
then just use `writev()` (https://linux.die.net/man/2/writev) to sequentially write the two buffers.
or what do you use to work with sockets?

**Comment by DarwinRepo** [comment]
shared memory, i use it just one PC two process. i need it faster

**Comment by niXman** [comment]
then use `memcpy()` for the two buffers.

**Comment by DarwinRepo** [comment]
it doesn't look elegant enough, i send size of the buffer first , then  use memcpy() for the two buffers.

**Comment by niXman** [comment]
yes, you are right. but the only elegant solution for me - to serialize the ID on the same level where you serialized other data...
any other solutions don't look elegant for me...

**Comment by niXman** [comment]
@DarwinRepo  if you have no other questions according this issue - please close the issue.
